backend/Controllers/ChessController.cs

```diff
@@ -32,6 +33,30 @@ public IActionResult CreateGame()
             games.Add(game);
             return Ok(new { GameId = game.Id}); 
         }
+        // GET: api/chess/{gameId}/moves
+        [HttpGet("{gameId}/moves")]
+        public IActionResult GetMovesHistory(string gameId)
+        {
+            var game = games.FirstOrDefault(g => g.Id.ToString() == gameId);
+            if (game == null)
+            {
+                return NotFound("Game not found.");
+            }
+            var moves = game.MovesArray;
+            if (game.MovesArray == null || !game.MovesArray.Any())
+            {
+                return Ok(new List<string>()); // Return an empty list if there are no moves
+            }
+            string jsonMoves = JsonSerializer.Serialize(moves);
+            MemoryStream memoryStream = new MemoryStream();
+            using (StreamWriter writer = new StreamWriter(memoryStream))
+            {
+                writer.Write(jsonMoves);
+                writer.Flush();
+            }
+            memoryStream.Position = 0;
+            return new FileStreamResult(memoryStream, "application/json");
+        }
 
         // POST: api/chessgame/{gameId}/move
         [HttpPost("{gameId}/move")]
```

Comment:
- i think it would be a good idea that this function would return game state or something similar. 
            - there is no way to check if the game has ended/its a draw/its a check